### PR TITLE
:feat listen to transactions for every adapter.

### PR DESCRIPTION
**Pull request by svenvoskamp**
# Description

Currently when a transaction has been made in AppKit, the balance won't update. This is a blocker for several clients. With this PR we are listening to transactionEvents and fetch the balance on these events.

wagmi: for wagmi we are using https://wagmi.sh/core/api/actions/watchPendingTransactions
ethers: for ethers we are using the `pending` event 
solana: For Solana we need to emit the event in the provider itself

## Type of change

- [ ] Chore (non-breaking change that addresses non-functional tasks, maintenance, or code quality improvements)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# Associated Issues

For Linear issues: Closes APKT-820
For GH issues: closes #...

# Showcase (Optional)

If there is a UI change include the screenshots with before and after state.
If new feature is being introduced, include the link to demo recording.

# Checklist

- [ ] Code in this PR is covered by automated tests (Unit tests, E2E tests)
- [ ] My changes generate no new warnings
- [ ] I have reviewed my own code
- [ ] I have filled out all required sections
- [ ] I have tested my changes on the preview link
- [ ] Approver of this PR confirms that the changes are tested on the preview link

### CODE REVIEW

File: packages/adapters/ethers/src/client.ts [hunk 1]
```diff
@@ -479,6 +480,19 @@ export class EthersAdapter extends AdapterBlueprint {
...
+  private listenPendingTransactions(provider: Provider) {
+    const browserProvider = new BrowserProvider(provider)
+
+    try {
+      browserProvider.on('pending', () => {
+        this.emit('pendingTransactions')
+      })
+    } catch (error) {
+      // eslint-disable-next-line no-console
+      console.error('Error listening to pending transactions', error)
```
Comment (lukaisailovic) [inline packages/adapters/ethers/src/client.ts:492]: Should we use `@walletconnect/logger` with new `somethingLogger` instance?

File: packages/appkit/src/client.ts [hunk 2]
```diff
@@ -1244,6 +1250,12 @@ export class AppKit {
       }
     })
 
+    adapter?.on('pendingTransactions', () => {
+      if (AccountController.state.address && ChainController.state.activeCaipNetwork?.id) {
+        this.updateBalance()
+      }
+    })
+
```
Comment (lukaisailovic) [inline packages/appkit/src/client.ts:1258]: Should we do optional chaining here? I wonder if there is a better way to handle this. We should check if undefined, or assert somewhere earlier in the flow

File: packages/adapters/ethers/src/client.ts (hunk 1 above)
Comment (magiziz) [inline packages/adapters/ethers/src/client.ts:492]: Maybe we can use [AlertController](https://github.com/reown-com/appkit/blob/main/packages/core/src/controllers/AlertController.ts) for users who use debug mode ?

Comment (svenvoskamp) [reply packages/adapters/ethers/src/client.ts:492]: Using AlertController as suggested by @magiziz

**Review by magiziz — APPROVED** [review]

**Pull request merged**